port_ocean/helpers/retry.py

```diff
@@ -132,13 +132,20 @@ def handle_request(self, request: httpx.Request) -> httpx.Response:
             httpx.Response: The response received.
 
         """
-        transport: httpx.BaseTransport = self._wrapped_transport  # type: ignore
-        if request.method in self._retryable_methods:
-            send_method = partial(transport.handle_request)
-            response = self._retry_operation(request, send_method)
-        else:
-            response = transport.handle_request(request)
-        return response
+        try:
+            transport: httpx.BaseTransport = self._wrapped_transport  # type: ignore
+            if request.method in self._retryable_methods:
+                send_method = partial(transport.handle_request)
+                response = self._retry_operation(request, send_method)
+            else:
+                response = transport.handle_request(request)
+            return response
+        except Exception as e:
+            if isinstance(e, httpx.RequestError) and not self._is_retryable_method(
+                request
```

Comment:
what about other errors? maybe worth logging every raised error when the request is not retryable?